### ISSUE TITLE
Hab Supervisor will accept messages for UUIDs that are not its own

### ISSUE THREAD

**Issue by thomascate** [habitat-sh/habitat#5865]
If a hab supervisor in a large cluster goes away, and within a short period of time another shows up with the same IP address with a different UUID, the new supervisor will join the old ones ring even if it is not configured to. 

The time period seems to be a factor of ring size as well. With a ring of about 100 nodes if the supervisor goes away and a new one gets the IP with seconds, the new one will join the old ring immediately and carry on like there was no issue. In a case where I took a supervisor out of the ring and waited 10 minutes before putting a new one in its place, it didn't immediately join the ring. However within about 30 minutes I got these messages for old UUIDs it used to have.
```
ERROR 2018-11-20T20:11:44Z: habitat_butterfly::server::inbound: PingReq request PingReq { membership: [Membership { member: Member { id: "9cffa808412f42b5bccf8843e908e077", incarnation: Incarnation(0), address: "10.8.16.73", swim_port: 9638, gossip_port: 9638, persistent: false, departed: false }, health: Alive }], from: Member { id: "85f349ebda1f4736953f7890cff38721", incarnation: Incarnation(0), address: "10.8.12.79", swim_port: 9638, gossip_port: 9638, persistent: false, departed: false }, target: Member { id: "9cffa808412f42b5bccf8843e908e077", incarnation: Incarnation(0), address: "10.8.16.73", swim_port: 9638, gossip_port: 9638, persistent: false, departed: false } } for invalid target
ERROR 2018-11-20T20:11:46Z: habitat_butterfly::server::inbound: PingReq request PingReq { membership: [Membership { member: Member { id: "cd3e8e67f5fd4631808a4a517ce7ba3e", incarnation: Incarnation(0), address: "10.8.16.76", swim_port: 9638, gossip_port: 9638, persistent: false, departed: false }, health: Alive }], from: Member { id: "46b82d398faa41e294035c1a049bee52", incarnation: Incarnation(0), address: "10.8.11.79", swim_port: 9638, gossip_port: 9638, persistent: false, departed: false }, target: Member { id: "cd3e8e67f5fd4631808a4a517ce7ba3e", incarnation: Incarnation(0), address: "10.8.16.76", swim_port: 9638, gossip_port: 9638, persistent: false, departed: false } } for invalid target
```
After which I curl'd the butterfly endpoint and saw that it had joined the ring.

The easiest way to recreate this is to spin up a decently sized ring with something like docker or kubernetes, then

Join a hab sup to your cluster,
`hab sup run --peer 10.8.1.53`

 verify it's working ok and sees cluster members
```
curl -s localhost:9631/butterfly | jq .member.members | grep \"id\" | wc -l
511
```

Stop hab sup and blow away the MEMBER_ID and rumor file, start hab sup with no peer
```
^Chab-sup(MR): Gracefully departing from butterfly network.
hab-launch(SV): Hasta la vista, services.
ERROR 2018-11-20T20:44:20Z: hab_launch: Launcher exiting with code 0

rm -rf /hab/sup/default/MEMBER_ID /hab/sup/default/data
hab sup run
```

Notice you're back in the cluster with a new UUID
```
curl -s localhost:9631/butterfly | jq .member.members | grep \"id\" | wc -l
512
```

I suspect it's receiving a rumor file for the old UUID, parses it and joins the cluster that the old one used to be a part of.

**Comment by christophermaier** [comment]
Did a bit of preliminary poking around and discovered that we don't seem to send the Member ID of the intended message recipient. 

We'd need to add that at the very least for the recipient to determine if a message is intended for itself or not. Whether that's done by adding a new Member struct or a bare Member ID field needs some investigation.

The fact that we won't have a Member ID when we're doing initial peering of a new Supervisor into an existing ring is a potentially complicating factor.

TL;DR - this isn't going to be the quick fix I had hoped it would be, and will require a bit more research and planning.

**Comment by christophermaier** [comment]
Still relevant.
